src/clients/InsightsQueryClient.ts

```diff
@@ -102,6 +102,17 @@ class InsightsQueryClient {
     return response.data as Insights.InsightsQueryResponse<Insights.QueryGetPoolsVolumeResponse>
   }
 
+  async CompetitionLeaderboard(req: Insights.QueryGetCompetitionLeaderboardRequest): Promise<Insights.InsightsQueryResponse<Insights.QueryGetCompetitionLeaderboardResponse>> {
+    const queryParams = {
+      market: req.market,
```

Comment:
For market param, I noticed if you pass an array of strings directly, the resulting query becomes `?market=<marketA>&market=<marketB>` and it does not filter. It would be better to convert `req.market` into a string (i.e. `?market=<marketA>,<marketB>`) before adding to the query params